sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -779,9 +782,14 @@ object IntervalUtils {
    * @throws ArithmeticException If numeric overflow occurs
    */
   def durationToMicros(duration: Duration): Long = {
-    val us = Math.multiplyExact(duration.getSeconds, MICROS_PER_SECOND)
-    val result = Math.addExact(us, duration.getNano / NANOS_PER_MICROS)
-    result
+    val seconds = duration.getSeconds
+    if (seconds == minDurationSeconds) {
+      val us = (minDurationSeconds + 1) * MICROS_PER_SECOND
+      Math.addExact(us, (duration.getNano - NANOS_PER_SECOND) / NANOS_PER_MICROS)
```

Comment:
is it possible that `duration.getNano - NANOS_PER_SECOND` overflows? Or `duration.getNano` is guaranteed to be positive?

Reply:
According to Java doc for `Duration.getNano`:
```java
     * @return the nanoseconds within the second part of the length of the duration, from 0 to 999,999,999
     */
    public int getNano() {
        return nanos;
    }
```

Reply:
It seems a mess at java-side already
```scala
scala> java.time.Duration.ofNanos(-1)
res6: java.time.Duration = PT-0.000000001S

scala> java.time.Duration.ofNanos(-1).get
get   getClass   getNano   getSeconds   getUnits

scala> java.time.Duration.ofNanos(-1).getNano
res9: Int = 999999999

scala> java.time.Duration.ofNanos(-1000).getNano
res10: Int = 999999000

scala> java.time.Duration.ofNanos(Int.MinValue).getNano
res12: Int = 852516352

scala> java.time.Duration.ofNanos(Int.M).getNano
MaxValue   MinValue

scala> java.time.Duration.ofNanos(Int.MaxValue).getNano
res15: Int = 147483647
```

Reply:
Why? `getNano` returns an adjustment to seconds, right?

Reply:
then we are safe. Can we add a comment or assert to explain that this won't overflow?

Reply:
ah， right